src/SignalR/server/Core/src/Internal/DefaultHubDispatcher.cs

```diff
@@ -342,17 +358,18 @@ async Task ExecuteInvocation()
             {
                 if (disposeScope)
                 {
-                    CleanupInvocation(connection, hubMethodInvocationMessage, hubActivator, hub, scope);
+                    await CleanupInvocation(connection, hubMethodInvocationMessage, hubActivator, hub, scope);
                 }
             }
         }
 
-        private void CleanupInvocation(HubConnectionContext connection, HubMethodInvocationMessage hubMessage, IHubActivator<THub> hubActivator,
+        private async Task CleanupInvocation(HubConnectionContext connection, HubMethodInvocationMessage hubMessage, IHubActivator<THub> hubActivator,
             THub hub, IServiceScope scope)
         {
             hubActivator?.Release(hub);
-            scope.Dispose();
 
+            await scope.DisposeAsync();
```

Comment:
I think it's safe to return the task at the end of the function so no statemachine.

```
var scopeDisposeTask = scope.DisposeAsync();
// ...
return scopeDisposeTask;
```

Reply:
Why?

Reply:
To get rid of the statemachine

Reply:
It doesn't matter here. Not even a little bit

Reply:
OK did it 👍